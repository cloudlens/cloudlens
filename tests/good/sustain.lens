var last = {};
function sustain (e) {
    for (prop in e) {
	    if (typeof e[prop] !== undefined) {
	        last[prop] = e[prop]
	    }
    }
    return res;
}

match {
    "^(?<name>[^ ]*) picked";
    "(?<n>\d+)"
}

stream when (e.message){
    sustain(e);
}

stream {
    print(last.name + " picked " + e.n + " fruits");
}
