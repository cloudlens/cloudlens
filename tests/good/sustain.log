Alice picked 2 apples
3 oranges
Nicolas picked
2 pears
5 bananas
Olivier picked nothing
Mandana picked 1 pineapples
