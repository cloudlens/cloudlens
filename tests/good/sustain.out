Alice picked 2 fruits
Alice picked 3 fruits
Nicolas picked 2 fruits
Nicolas picked 5 fruits
Mandana picked 1 fruits
